plugins/org.eclipse.elk.core.service/src/org/eclipse/elk/core/service/graphvisitors/CompoundGraphElementVisitor.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0  *******************************************************************************/
+package org.eclipse.elk.core.service.graphvisitors;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.elk.core.util.ElkUtil;
+import org.eclipse.elk.core.util.IGraphElementVisitor;
+import org.eclipse.elk.graph.ElkGraphElement;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Class to apply many graph element visitors on the same graph.
+ */
+public class CompoundGraphElementVisitor implements IGraphElementVisitor {
+    
+    /**
+     * List of graph visitors which will be applied in the given order.
+     */
+    private List<IGraphElementVisitor> graphVisitors = new LinkedList<>();
+    
+    /**
+     * Creates a compound graph visitor using the given visitors.
+     * 
+     * @param visitors The graph visitors
+     */
+    public CompoundGraphElementVisitor(final IGraphElementVisitor... visitors) {
+        graphVisitors.addAll(Arrays.asList(visitors));
+    }
+
+    /* (non-Javadoc)
+     * @see org.eclipse.elk.core.util.IGraphElementVisitor#visit(org.eclipse.elk.graph.ElkGraphElement)
+     */
+    @Override
+    public void visit(final ElkGraphElement element) {
+        if (element.eContainer() == null && element instanceof ElkNode) {
+            ElkUtil.applyVisitors((ElkNode) element,
+                    graphVisitors.toArray(new IGraphElementVisitor[graphVisitors.size()]));
+        }
+
```

Comment:
Hm... there's an interesting issue here. This class could operate in two ways:

1. Just forward each call to its `visit(...)` method to its child visitors.
1. Have each visitor visit the entire graph before applying the next visitor. (this is the current behaviour)

I think both should be supported, and the first should be the default since its faster.

Reply:
Another problem here is that the visitors are only applied on the root node (`element.eContainer() == null`). I will change to your suggested default behavior with the option to configure it to the (corrected) current behavior.